consensus/consensus-types/src/payload_pull_params.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{
+    common::{Author, PayloadFilter},
+    utils::PayloadTxnsSize,
+};
+use std::{collections::HashSet, time::Duration};
+
+#[derive(Clone)]
+pub struct OptQSPayloadPullParams {
+    pub opt_batch_txns_pct: u8,
```

Comment:
If we saturate TPS instantly by flooding mempool with many transactions, there will be more proofs than can fit in a block and we will never be in a situation where we will work through our proof backlog and start pulling opt batches.

Reply:
isn't that expected behavior in some sense? if we pull the opt batches, we delay the existing proofs even further. if we delay the opt batch, it may become proof in next block. plus the opt batch may contain txns that depend on txns from existing proofs

Reply:
yes, prioritizing will delay proofs and they might expire. it's an attempt to force more opt batches under this situation and reach equlibirum. Otherwise, we will always be proposing proofs. Of course, if backlog far exceeds the number of opt batches that can fit in a block, then we will have a backlog of both proofs and opt batches. And yes, if there is a dependency between opt batch and proofs, we cannot execute. I am just worried in case of sudden spikes, we might not get the savings we expect.

Reply:
This is not a blocker. I can remove this and we can experiment it later as well.

Reply:
Removed this behavior for now. I can experiment further and we can reintroduce later if necessary.